akka-docs/rst/java/stream/stream-quickstart.rst

```diff
@@ -1,7 +1,140 @@
+called a :class:`Sink`.
+
+Reusable Pieces
+---------------
+
+One of the nice parts of Akka Streams—and something that other stream libraries
+do not offer—is that not only sources can be reused like blueprints, all other
+elements can be as well. We can take the file-writing :class:`Sink`, prepend
+the processing steps necessary to get the :class:`ByteString` elements from
+incoming strings and package that up as a reusable piece as well. Since the
+language for writing these streams always flows from left to right (just like
+plain English), we need a starting point that is like a source but with an
+“open” input. In Akka Streams this is called a :class:`Flow`:
+
+.. includecode:: ../code/docs/stream/QuickStartDocTest.java#transform-sink
+
+Starting from a flow of strings we convert to :class:`ByteString` and then feed
+to the already known file-writing :class:`Sink`. The resulting blueprint is a
+:class:`Sink<String, CompletionStage<IOResult>>`, which means that it accepts
+strings as its input and when materialized it will create auxiliary information
+of type ``CompletionStage<IOResult>`` (when chaining operations on a
+:class:`Source` or :class:`Flow` the type of the auxiliary information—called
+the “materialized value”—is given by the leftmost starting point; since we want
+to retain what the ``FileIO.toFile`` sink has to offer, we need to say
+``Keep.right()``). :class:`IOResult` is a type that IO operations return in
+Akka Streams in order to tell you how many bytes or elements were consumed and
+whether the stream terminated normally or exceptionally.
+
+We can use the new and shiny :class:`Sink` we just created by
+attaching it to our ``factorials`` source—after a small adaptation to turn the
+numbers into strings:
+
+.. includecode:: ../code/docs/stream/QuickStartDocTest.java#use-transformed-sink
+
+Time-Based Processing
+---------------------
+
+Before we start looking at a more involved example we explore the streaming
+nature of what Akka Streams can do. Starting from the ``factorials`` source
+we transform the stream by zipping it together with another stream,
+represented by a :class:`Source` that emits the number 0 to 100: the first
+number emitted by the ``factorials`` source is the factorial of zero, the
+second is the factorial of one, and so on. We combine these two by forming
+strings like ``"3! = 6"``.
+
+.. includecode:: ../code/docs/stream/QuickStartDocTest.java#add-streams
+
+All operations so far have been time-independent and could have been performed
+in the same fashion on strict collections of elements. The next line
+demonstrates that we are in fact dealing with streams that can flow at a
+certain speed: we use the ``throttle`` combinator to slow down the stream to 1
+element per second (the second ``1`` in the argument list is the maximum size
+of a burst that we want to allow—passing ``1`` means that the first element
+gets through immediately and the second then has to wait for one second and so
+on). 

```

Comment:
I’d rather keep it as concise as possible, the purpose is explained and the name is quite fitting, so I don’t think this is a hindrance to understanding.